### PR TITLE
TypeScript conversion

### PR DESCRIPTION
**Pull request by muan**
- TypeScript
- Remove UMD build
- Update test to run on multiple OSes
- Update dependencies
- Remove Travis CI config

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -1,13 +0,0 @@
...
-  disabled: boolean
-  sortable: boolean
-}
-
-declare global {
-  interface Window {
-    TaskListsElement: typeof TaskListsElement
-  }
-  interface HTMLElementTagNameMap {
-    'task-lists': TaskListsElement
```
Comment (muan) [inline index.d.ts:11]: @koddsson do you have context on this addition? Looks like by default tsc does not generate these lines.

Comment (koddsson) [reply index.d.ts:11]: Yup. Context is [here](https://github.com/github/image-crop-element/pull/26).

File: package.json [hunk 2]
```diff
@@ -3,16 +3,17 @@
...
+  "main": "dist/index.js",
+  "module": "dist/index.js",
+  "type": "module",
+  "types": "dist/index.d.ts",
   "scripts": {
     "clean": "rm -rf dist",
     "lint": "github-lint",
-    "prebuild": "npm run clean && npm run lint",
-    "build": "rollup -c && cp src/task-lists-element.js.flow dist/index.esm.js.flow && cp src/task-lists-element.js.flow dist/index.umd.js.flow",
+    "prebuild": "npm run clean && npm run lint && tsc",
```
Comment (koddsson) [inline package.json:13]: Should this have `--noEmit`?

```suggestion
    "prebuild": "npm run clean && npm run lint && tsc --noEmit",
```

Comment (muan) [reply package.json:13]: The is generating ds and compiling js to ts for rollup.

File: index.d.ts (hunk 1 above)
Comment (muan) [reply index.d.ts:11]: Added them in https://github.com/github/task-lists-element/pull/16/commits/641ba8f247424a400ec1c7e921b9e0dfef965440.

**Pull request merged**